utils/webpack/plugins/ShopgateIndexerPlugin.js

```diff
@@ -0,0 +1,387 @@
+
+    if (!componentExists(componentPath)) {
+      return;
+    }
+
+    const variableName = getVariableName(id);
+
+    const isPortalsOrWidgets = (
+      (type === TYPE_PORTALS && component.target !== 'app.routes')
+      || type === TYPE_WIDGETS
+    );
+
+    if (isPortalsOrWidgets && has(themePackage.dependencies, 'react-loadable')) {
+      imports.push(`const ${variableName} = Loadable({\n  loader: () => import('${componentPath}'),\n  loading: Loading,\n});\n`);
+    } else {
+      imports.push(`import ${variableName} from '${componentPath}';`);
+    }
+
+    if (isArray) {
+      exports.push(`  ${variableName},`);
+      return;
+    }
+
+    exports.push(`  '${id}': ${variableName},`);
+  });
+
+  if (importsEnd) {
+    imports.push(importsEnd);
+  }
+
+  exports.push(exportsEnd);
+
+  return {
+    imports,
+    exports,
+  };
+}
+
+/**
+ * Creates translations for the extension indexing log.
+ * @param {string} type The indexed type.
+ * @returns {Object}
+ */
+function getIndexLogTranslations(type = TYPE_WIDGETS) {
+  const params = { type: t(`TYPE_${type}`) };
+
+  return {
+    logStart: `  ${t('INDEXING_TYPE', params)}`,
+    logEnd: `  ${t('INDEXED_TYPE', params)}`,
+    logNotFound: `  ${t('NO_EXTENSIONS_FOUND_FOR_TYPE', params)}`,
+  };
+}
+
+/**
+ * Validates the extensions input.
+ * @param {Object} input The input.
+ * @return {Promise}
+ */
+function validateExtensions(input) {
+  try {
+    const extensionPath = getExtensionsPath();
+
+    if (!fs.existsSync(extensionPath)) {
+      fs.mkdirSync(extensionPath);
+    }
+
+    if (!input.imports.length) {
+      return null;
+    }
+
+    return input;
+  } catch (e) {
+    throw new Error(t('EXTENSION_COULD_NOT_BE_VALIDATED'));
+  }
+}
+
+/**
+ * Creates the string that is written into the appropriate file.
+ * @param {Object} input The input object.
+ * @return {string}
+ */
+function createStrings(input) {
+  try {
+    if (!input) {
+      return null;
+    }
+
+    const importsString = input.imports.length ? `${input.imports.join('\n')}\n\n` : '';
+    const exportsString = input.exports.length ? `${input.exports.join('\n')}\n` : '';
+    const indexString = `${importsString}${exportsString}`.replace('\n\n\n', '\n\n');
+
+    return indexString.length ? indexString : null;
+  } catch (e) {
+    throw new Error(t('STRINGS_COULD_NOT_BE_CREATED'));
+  }
+}
+
+/**
+ * Writes to extension file.
+ * @param {Object} options The action object.
+ */
+function writeExtensionFile(options) {
+  const {
+    file,
+    input,
+    defaultContent,
+    logNotFound,
+    logEnd,
+  } = options;
+  const filePath = path.resolve(getExtensionsPath(), file);
+
+  if (!input) {
+    logger.warn(logNotFound);
+    fs.writeFileSync(filePath, defaultContent, { flag: 'w+' });
+    return;
+  }
+
+  fs.writeFileSync(filePath, input, { flag: 'w+' });
+  logger.log(logEnd);
+}
+
+/**
+ * Creates an index.
+ * @param {Object} options The indexing options,
+ */
+function index(options) {
```

Comment:
Portals are not appended to theme. You can check it with @shopgate-product-reviews extension. RatingStars portals are not written into `portals.js file`

Reply:
Fixed.